marklogic-data-hub/src/test/java/com/marklogic/hub/hubcentral/InitializedDataHubProject.java

```diff
@@ -0,0 +1,96 @@
+    private File zipProjectFile;
+
+    public InitializedDataHubProject(HubClient hubClient) {
+        super(hubClient);
+    }
+
+    public void writeProjectToZipFile() {
+        try {
+            zipProjectFile = new File("build/allProject.zip");
+            FileOutputStream fos = new FileOutputStream(zipProjectFile);
+            new HubCentralManager().writeProjectAsZip(hubClient, fos);
+            fos.close();
+            readZipProject();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void readZipProject() throws IOException {
+        zipProjectEntries = new HashMap<>();
+        zipArtifactEntries = new HashMap<>();
+        ZipFile zip = new ZipFile(zipProjectFile);
+        Enumeration<?> entries = zip.entries();
+        String[] artifactDirs = {"flows", "steps", "entities", "src/main/entity-config", "src/main/ml-config/security/protected-paths",
+            "src/main/ml-config/security/query-rolesets"};
+        while (entries.hasMoreElements()) {
+            ZipEntry entry = (ZipEntry) entries.nextElement();
+            if(Stream.of(artifactDirs).anyMatch(entry.getName()::startsWith) && !entry.isDirectory()){
+                readArtifactZipEntry(zip, entry);
+            }
+            else{
+                String key;
+                if(!entry.isDirectory()){
+                    assertTrue(entry.getSize() > 0);
+                    int index = entry.getName().lastIndexOf('/');
+                    if(index == -1){
+                        key = "rootDir";
+                    }
+                    else{
+                        key = entry.getName().substring(0, entry.getName().lastIndexOf('/'));
+                    }
+                    zipProjectEntries.computeIfPresent(key, (k, v) -> v + 1);
+                }
+                else{
+                    key = entry.getName();
+                }
+                zipProjectEntries.putIfAbsent(key, 1);
+            }
+        }
+    }
+
+    public void verifyZipProject(){
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/custom/"));
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/ingestion/"));
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/mapping/"));
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/mapping-functions/"));
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/mastering/"));
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/matching/"));
+        assertTrue(zipProjectEntries.containsKey("src/main/ml-modules/root/custom-modules/merging/"));
+        assertEquals(5, zipProjectEntries.get("rootDir"));
+        assertEquals(2, zipProjectEntries.get("src/main/hub-internal-config/database-fields"));
```

Comment:
These assertions on exact counts create a maintenance burden, as any time someone adds a new file to hub-internal-config, they have to remember to update this test - which is very unlikely that someone would think to do that, which means this test will fail. And the developer will have to figure out what's going on in this test and how to fix it. 

Instead of exact counts that are brittle, I think we should just verify that one file in each directory exists. That should suffice for verifying that hub-internal-config was written out correctly. 

Also, by not depending on brittle counts, I don't think you need the additional logic above for making counts of files in directories.